### DISCUSSION TITLE
Full Lambda Syntax without `=`

### DISCUSSION THREAD

**Discussion by msadeqhe** [hsutter/cppfront#793] [Suggestions]
Currently Cpp2 has this terser lambda syntax:

```cpp
: (x) x + 1
```

Would it be possible to also drop `=` from full lambda syntax?

```cpp
: (x: i32) -> i32 { return x + 1; }

: (x: i32) { print(x); }
```

Consequently function declaration syntax would be changed too:

```cpp
fnc1: (x: i32) -> i32
{
    return x + 1;
}

fnc2: (x: i32)
{
    print(x);
}
```

Literally for consistency, it leads to dropping `=` from all other declarations too:

```cpp
spc1: namespace
{
    cls1: type
    {
        fld1: i32 = 10;
    }
}
```

So all declarations would be less divergent from Cpp1 as discussed in #742.

**Comment by SebastianTroy** [comment]
Also using = in namespaces declarations makes it feel like a subsequent declaration should replace the name space content, rather than extend it.


On 31 October 2023 08:23:16 Sadeq ***@***.***> wrote:

Currently Cpp2 has this terser lambda syntax:

: (x) x + 1

Would it be possible to also drop = from full lambda syntax?

: (x: i32) -> i32 { return x + 1; }

Consequently function declaration syntax would be changed too:

fnc1: (x: i32) -> i32
{
        return x + 1;
}

Literally for consistency, it leads to dropping = from all other declarations too:

spc1: namespace
{
        cls1: type
        {
                fld1: i32 = 10;
        }
}

So all declarations would be less divergence from Cpp1 as discussed in #742<https://github.com/hsutter/cppfront/discussions/742>.

**Reply by JohelEGP** [reply]
That's why I opened #634.
